java/src/jmri/jmrit/operations/trains/TrainsTableModel.java

```diff
@@ -595,7 +597,7 @@ public void propertyChange(PropertyChangeEvent e) {
             synchronized (this) {
                 SwingUtilities.invokeLater(() -> {
                     int row = sysList.indexOf(train);
```

Comment:
If it's `sysList` that needs protection, then I suggest that you synchronize this line only:

```
int row;
synchronized (this) {
    row = sysList.indexOf(train);
}
```